ouroboros-consensus/src/ouroboros-consensus/Ouroboros/Consensus/MiniProtocol/BlockFetch/ClientInterface.hs

```diff
@@ -295,6 +295,21 @@ mkBlockFetchConsensusInterface
           -- fragment can't be empty.
           (True,  _)     -> error "impossible"
 
+      -- The precondition of 'compareAnchoredFragments' is that either both
+      -- fragments are non-empty or they intersect. We are mostly satisfying
```

Comment:
The "either" here and also in the original precondition is actually slightly awkward as we want inclusive, not exclusive or. But I don't think it is too misleading, so it could stay as is.

Reply:
Good point -- I'll remove the "either"